torch/fx/experimental/symbolic_shapes.py

```diff
@@ -648,6 +648,53 @@ def check_supported_type(x):
                 sympy.simplify(base / gcd), sympy.simplify(divisor / gcd)
             )
 
+class Mod(sympy.Function):
+    """
+    Mod(a, b) => a % b
+    """
+
+    nargs = (2,)
+    is_integer = True
+    precedence = 50
+
+    def _sympystr(self, printer):
+        base = printer.parenthesize(self.args[0], self.precedence)
+        divisor = printer.parenthesize(self.args[1], self.precedence)
+        return f"{base}%{divisor}"
+
+    @classmethod
+    def eval(cls, base, modulus):
+        if base == 0 or modulus == 1:
+            return sympy.Integer(0)
+
+        if (
+            isinstance(base, sympy.Integer)
+            and isinstance(modulus, sympy.Integer)
+        ):
+            return base % modulus
+
+        if isinstance(base, sympy.Add):
```

Comment:
sympy has a comment like

```
        # (x + y + 2) % x -> Mod(y + 2, x)
```

which I thought was pretty useful for understanding what their Add rule is simplifying for